internal/workload/manager.go

```diff
@@ -234,6 +234,82 @@ func (w *WorkloadManager) RegisterObserver(observer Observer) {
 	w.workloads.RegisterObserver(observer)
 }
 
+func (w *WorkloadManager) Deregister() error {
+	w.managementLock.Lock()
+	defer w.managementLock.Unlock()
+
+	err := w.removeAllWorkloads()
+	if err != nil {
+		log.Errorf("failed to remove workloads: %v", err)
+	}
+
+	err = w.deleteManifestsDir()
+	if err != nil {
+		log.Errorf("failed to delete manifests directory: %v", err)
+	}
+
+	err = w.deleteTable()
+	if err != nil {
+		log.Errorf("failed to delete table: %v", err)
+	}
+
+	err = w.deleteVolumeDir()
```

Comment:
@jakub-dzon wouldn't we like to make sure datatransfer/monitor.syncPaths() is completed before this step to avoid data loss ?

Reply:
Good point.  We need to make at least one sync after all the workloads have been stopped.

Reply:
As I understood - after removing each workloads there is a call to WorkloadRemoved for each observer
So I think that no need to add "sync" call... 
@masayag @jakub-dzon what do you think?

Reply:
As I understood - after removing each workloads there is a call to WorkloadRemoved for each observer
So I think that no need to add "sync" call...
@masayag  @jakub-dzon what do you think?

Reply:
Yes, there doesn't need to be an explicit call to sync data, but `WorkloadRemoved` method needs to be modified to execute that one last sync for the workload.